openweathermap-ios/openweathermap-ios/MapViewController.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  MapViewController.swift
+//  openweathermap-ios
+//
+
+
+import UIKit
+import MapKit
+
+final class MapViewController: UIViewController, CLLocationManagerDelegate {
+
+    @IBOutlet private weak var mapView: MKMapView!
+    private let locationManager = CLLocationManager()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        mapView.delegate = self
+        locationManager.delegate = self
+        locationManager.desiredAccuracy = kCLLocationAccuracyBest
+        locationManager.requestWhenInUseAuthorization()
+        locationManager.startUpdatingLocation()
+        
+    }
+    
+    
+    func addAnnotation(location: CLLocationCoordinate2D){
+        let circleCenter = MKCircle(center: location, radius: 5)
+        let circleEdge = MKCircle(center: location, radius: 50)
+        
+        mapView.overlays.forEach { overlay in
+            if overlay is MKCircle{
+                mapView.removeOverlay(overlay)
+            }
+        }
+        mapView.addOverlay(circleCenter)
+        mapView.addOverlay(circleEdge)
+    }
+}
+
+extension MapViewController: MKMapViewDelegate {
+    func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus) {
+        if status != .authorizedWhenInUse {
+            let localizedTitleAlertController = NSLocalizedString("titleUIAlertController", comment: "")
+            let localizedContentAlertController = NSLocalizedString("messageUIAlertController", comment: "")
```

Comment:
```suggestion
            let title = NSLocalizedString("titleUIAlertController", comment: "")
            let message = NSLocalizedString("messageUIAlertController", comment: "")
```

You don't need to specify too much the constant.

Reply:
Review the other constant names too.

Reply:
I can change the first two constants. 
Both are within the same code block
`let title = NSLocalizedString ("titleUIAlertController", comment: "")`
 `let message = NSLocalizedString ("messageUIAlertController", comment: "")`
But the second title cannot be just "title" because it is another constant. Is there any way to leave the two constants "title"?
 `let titleSettings = NSLocalizedString ("settingsUIAlertAction", comment: "")`

Reply:
Go with:

alertTitle, alertMessage, settingsTitle, etc..

Reply:
Remove all that doesn't add value and keep only the necessary to give enough context.